### PR TITLE
Fix SDL fails TLS handshake.

### PR DESCRIPTION
**Pull request by indlu**
Fixed SSL context getting and encrypted service starting.

### CODE REVIEW

**Comment by indlu** [comment]
@LuxoftAKutsan, please review

File: src/components/protocol_handler/src/protocol_handler_impl.cc [hunk 1]
```diff
@@ -1027,8 +1027,12 @@ class StartSessionHandler : public security_manager::SecurityManagerListener {
         protocol_handler_->SendStartSessionNAck(
             connection_id_, session_id_, protocol_version_, service_type_);
       } else {
-        // Could not be success handshake and not already protected service
-        NOTREACHED();
+        protocol_handler_->SendStartSessionAck(connection_id_,
```
Comment (LuxoftAKutsan) [inline src/components/protocol_handler/src/protocol_handler_impl.cc]: @indlu you sure this DCHEK shuld be removed?

File: src/components/protocol_handler/src/protocol_handler_impl.cc [hunk 2]
```diff
@@ -1120,7 +1124,9 @@ RESULT_CODE ProtocolHandlerImpl::HandleControlMessageStartSession(
                                                   connection_key);
     security_manager::SSLContext* ssl_context =
         GetSSLContextBySession(session_info);
-
+    if (!ssl_context) {
+      ssl_context = security_manager_->CreateSSLContext(connection_key);
```
Comment (LuxoftAKutsan) [inline src/components/protocol_handler/src/protocol_handler_impl.cc]: @indlu create SSL context does the smae as GetSSL context. Difference in setting protection flag. 

Calling GetSSLContextBySession and then CreateSSLContext looks weired. I suppose it is not correct.

Comment (LuxoftAKutsan) [inline src/components/protocol_handler/src/protocol_handler_impl.cc]: @indlu seting protection flag should be executed when we sure that handshake done shouldn't we?

File: src/components/protocol_handler/src/protocol_handler_impl.cc [hunk 3]
```diff
@@ -1231,7 +1237,35 @@ RESULT_CODE ProtocolHandlerImpl::HandleControlMessageStartSession(
...
+                                  packet.protocol_version(),
+                                  si.hash_id_,
+                                  service_type,
+                                  get_settings().force_protected_service()));
+      if (!ssl_context->IsHandshakePending()) {
+        // Start handshake process
+        security_manager_->StartHandshake(connection_key);
+      }
+    }
+    LOG4CXX_DEBUG(logger_,
```
Comment (LuxoftAKutsan) [inline src/components/protocol_handler/src/protocol_handler_impl.cc]: @indlu I suppose this debug is related to else branch upper?

**Comment by indlu** [comment]
Rebuild required

**Comment by indlu** [comment]
Rebuild required

**Comment by indlu** [comment]
Rebuild required

**Pull request merged**